pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/dictionary/DictionaryIndexType.java

```diff
@@ -348,6 +353,50 @@ protected Dictionary createIndexReader(PinotDataBuffer dataBuffer, ColumnMetadat
   @Override
   protected void handleIndexSpecificCleanup(TableConfig tableConfig) {
     IndexingConfig indexingConfig = tableConfig.getIndexingConfig();
+    List<String> noDictionaryColumns = indexingConfig.getNoDictionaryColumns() == null
+        ? Lists.newArrayList()
+        : indexingConfig.getNoDictionaryColumns();
+    List<FieldConfig> fieldConfigList = tableConfig.getFieldConfigList() == null
+        ? Lists.newArrayList()
+        : tableConfig.getFieldConfigList();
+
+    List<FieldConfig> configsToUpdate = new ArrayList<>();
+    for (FieldConfig fieldConfig : fieldConfigList) {
+      // skip further computation of field configs which already has RAW encodingType
+      if (fieldConfig.getEncodingType() == FieldConfig.EncodingType.RAW) {
+        continue;
+      }
+      // ensure encodingType is RAW on noDictionaryColumns
+      if (noDictionaryColumns.remove(fieldConfig.getName())) {
+        configsToUpdate.add(fieldConfig);
+      }
+      try {
+        DictionaryIndexConfig indexConfig = new ObjectMapper()
+            .treeToValue(fieldConfig.getIndexes().get(getPrettyName()), DictionaryIndexConfig.class);
+        // ensure encodingType is RAW where dictionary index config has disabled = true
+        if (indexConfig.isDisabled()) {
+          configsToUpdate.add(fieldConfig);
+        }
+      } catch (JsonProcessingException | NullPointerException ignore) {
```

Comment:
Why do we ignore the exception here? I would say that we need to throw it. Is there something I'm missing?

Reply:
Let me fix this. I was assuming there will never be a JsonProcessingException exception as the index config is code generated and purposefully ignored the NPE when there are no indexes.